src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -161,3 +173,113 @@ export const handleRedirect: RequestHandler<
     redirectPath,
   })
 }
+
+/**
+ * Handler for GET /:formId/publicform endpoint
+ * @returns 200 if the form exists
+ * @returns 404 if form with formId does not exist or is private
+ * @returns 410 if form has been archived
+ * @returns 500 if database error occurs or if the type of error is unknown
+ */
+export const handleGetPublicForm: RequestHandler<{ formId: string }> = async (
+  req,
+  res,
+) => {
+  const { formId } = req.params
+
+  const formResult = await getFormIfPublic(formId).andThen((form) =>
+    FormService.checkFormSubmissionLimitAndDeactivateForm(form),
+  )
+
+  // Early return if form is not public or any error occurred.
+  if (formResult.isErr()) {
+    const { error } = formResult
+    logger.error({
+      message: 'Error retrieving public form',
+      meta: {
+        action: 'handleGetPublicForm',
+        ...createReqMeta(req),
+        formId,
+      },
+      error,
+    })
+    const { errorMessage, statusCode } = mapRouteError(error)
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+
+  const form = formResult.value
+  const publicForm = form.getPublicView()
+  const { authType } = form
+  let myInfoError = false
+
+  // NOTE: Creating a variable to ensure that TS will enforce the type and ensure all keys in AuthType are covered.
+  const formController: FormController<
+    | ReturnType<typeof SpcpFactory.createFormWithSpcpSession>
+    | ReturnType<typeof MyInfoFactory.createFormWithMyInfo>
+  > = {
+    [AuthType.SP]: () =>
+      SpcpFactory.createFormWithSpcpSession(form, req.cookies),
+    [AuthType.CP]: () =>
+      SpcpFactory.createFormWithSpcpSession(form, req.cookies),
+    [AuthType.MyInfo]: () => {
+      return MyInfoFactory.createFormWithMyInfo(form, req.cookies)
+        .andThen((publicForm) => {
+          return extractSuccessfulMyInfoCookie(req.cookies).map(
+            (myInfoCookie) => {
+              const cookiePayload: MyInfoCookiePayload = {
+                ...myInfoCookie,
+                usedCount: myInfoCookie.usedCount + 1,
+              }
+              // NOTE: This is a side effect to set the cookie on the result after it has been successfully prefilled.
+              res.cookie(
+                MYINFO_COOKIE_NAME,
+                cookiePayload,
+                MYINFO_COOKIE_OPTIONS,
+              )
+              return publicForm
+            },
+          )
+        })
+        .mapErr((error) => {
+          // NOTE: This is a side-effect as there is no need for cookie if data could not be retrieved
+          res.clearCookie(MYINFO_COOKIE_NAME, MYINFO_COOKIE_OPTIONS)
+          // NOTE: This is done as a workaround because the type of FormController enforces a uniform return value
+          // but we are required to signal if myInfoError in the event of an error
+          myInfoError = true
```

Comment:
I don't think we should do this. We should instead use neverthrow's [`orElse`](https://www.npmjs.com/package/neverthrow#resultorelse-method) method to map errors back to an ok result, but with myInfoError set to true.

```ts
.orElse((error) => {
  // NOTE: This is a side-effect as there is no need for cookie if data could not be retrieved. 
  res.clearCookie(MYINFO_COOKIE_NAME, MYINFO_COOKIE_OPTIONS)
  return ok({
    ...publicForm,
    myInfoError: true,
  })
}
```

This will also remove the need to call `FormService.setMyInfoError(publicFormView, myInfoError),` (which actually will never be called if `myInfoError` is actually true since the chain will shortcircuit into the `mapErr` state.)